processor/src/trace/virtual_bus/sum_check/mod.rs

```diff
@@ -0,0 +1,277 @@
+        let virtual_poly = ProjectionComposition::new(0);
+
+        // Prover
+        let prover = SumCheckProver::new(virtual_poly);
+        let mut coin = RpoRandomCoin::new(Word::default());
+        let proof = prover.prove(claim, &mut mls, num_variables, &mut coin).unwrap();
+
+        // Verifier
+        let plain_query_builder = PlainQueryBuilder;
+        let verifier = SumCheckVerifier::new(virtual_poly, plain_query_builder);
+        let mut coin = RpoRandomCoin::new(Word::default());
+        let result = verifier.verify(claim, proof, &mut coin);
+
+        assert!(result.is_ok())
+    }
+
+    #[test]
+    fn test_sum_check_product() {
+        let num_variables = 14;
+        let values_0 = rand_vector(1 << num_variables);
+        let values_1 = rand_vector(1 << num_variables);
+        let claim = values_0.iter().zip(values_1.iter()).fold(ZERO, |acc, (x, y)| *x * *y + acc);
+
+        let ml_0 = MultiLinear::new(values_0.to_vec()).expect("should not fail");
+        let ml_1 = MultiLinear::new(values_1.to_vec()).expect("should not fail");
+        let mut mls = vec![ml_0, ml_1];
+        let virtual_poly = ProductComposition;
+
+        // Prover
+        let prover = SumCheckProver::new(virtual_poly);
+        let mut coin = RpoRandomCoin::new(Word::default());
+        let proof = prover.prove(claim, &mut mls, num_variables, &mut coin).unwrap();
+
+        // Verifier
+        let plain_query_builder = PlainQueryBuilder;
+        let verifier = SumCheckVerifier::new(virtual_poly, plain_query_builder);
+        let mut coin = RpoRandomCoin::new(Word::default());
+        let result = verifier.verify(claim, proof, &mut coin);
+
+        assert!(result.is_ok())
+    }
+
+    #[test]
+    fn test_sum_check_product_failure() {
+        let num_variables = 14;
+        let values_0 = rand_vector(1 << num_variables);
+        let values_1 = rand_vector(1 << num_variables);
+        let mut claim =
+            values_0.iter().zip(values_1.iter()).fold(ZERO, |acc, (x, y)| *x * *y + acc);
+
+        // modifying the claim should make the Verifier reject the proof
+        claim += ONE;
+
+        let ml_0 = MultiLinear::new(values_0.to_vec()).expect("should not fail");
+        let ml_1 = MultiLinear::new(values_1.to_vec()).expect("should not fail");
+        let mut mls = vec![ml_0, ml_1];
+        let virtual_poly = ProductComposition;
+
+        // Prover
+        let prover = SumCheckProver::new(virtual_poly);
+        let mut coin = RpoRandomCoin::new(Word::default());
+        let proof = prover.prove(claim, &mut mls, num_variables, &mut coin).unwrap();
+
+        // Verifier
+        let plain_query_builder = PlainQueryBuilder;
+        let verifier = SumCheckVerifier::new(virtual_poly, plain_query_builder);
+        let mut coin = RpoRandomCoin::new(Word::default());
+        let result = verifier.verify(claim, proof, &mut coin);
+
+        assert!(result.is_err())
+    }
+
+    struct PlainQueryBuilder;
```

Comment:
```suggestion
    struct ProjectionQueryBuilder;
```

Since this is the case where `f_i` returns the `i`th input